Assignment 2.Rmd

```diff
@@ -67,33 +77,175 @@ library(igraph)
 
 #First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
 
-g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+graph1 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
 
 #Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
 
-plot(g,layout=layout.fruchterman.reingold)
+plot(graph1,layout=layout.fruchterman.reingold)
 
 #There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
 
-plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+plot(graph1,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 #We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
 
-plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+plot(graph1,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+#remove clutter
+rm(V.STD)
+rm(V.TCH)
+
+#this way works
+#make new variable that counts total number of incidents by teacher and student from EDGE
+temp1 <- aggregate(tid ~ stid,FUN = length, data = D2)
+temp2 <- aggregate(stid ~ tid,FUN = length, data = D2)
```

Comment:
Another way to do this is to use the `group_by` and `summarize(sum(count))` functions for both students and teachers